### PR TITLE
Get data without login

### LINKED ISSUE #33: Anonymous access
Is there an option for not passing the password? I would like only to have access to public data

### PR DESCRIPTION
**Pull request by trunone**
resolve #33

### CODE REVIEW

File: pysma/__init__.py [hunk 1]
```diff
@@ -154,7 +154,11 @@ def __len__(self):
 
     def __contains__(self, key):
         """Check if a sensor is defined."""
-        return key in self
+        try:
```
Comment (kellerza) [inline pysma/__init__.py:157]: Lets revert the change to this function

File: pysma/__init__.py [hunk 2]
```diff
@@ -316,10 +328,14 @@ async def read(self, sensors):
...
         return True
 
-    async def read_logger(self, sensors, start, end):
+    async def read_logger(self, sensors=None, start=None, end=None):
         """Read a logging key and return the results."""
-        payload = {"destDev": [], "key": int(sensors[0].key), "tStart": start, "tEnd": end}
-        result_body = await self._read_body(URL_LOGGER, payload)
+        if self._new_session_data is None:
+            payload = {"destDev": [], "key": []}
+            result_body = await self._read_body(URL_DASH_LOGGER, payload)
```
Comment (kellerza) [inline pysma/__init__.py:335]: Does the Dashboard logger not support time ranges? Or is it just a simple view?

**Review by kellerza — COMMENTED** [review]
Thanks @trunone !

File: pysma/__init__.py (hunk 2 above)
Comment (trunone) [reply pysma/__init__.py:335]: Actually I'm not sure. I checked in DevTools. The "key' in request to /dyn/getDashLogger.json is empty. That's why I use empty key. Maybe time range is also working.

File: pysma/__init__.py (hunk 1 above)
Comment (trunone) [reply pysma/__init__.py:157]: OK, but this code will cause infinite recursions in my test. That's why I changed it to previous code. Maybe I did something wrong.

Comment (kellerza) [reply pysma/__init__.py:157]: Strange indeed. With your code, if the key exists, but the value of self[key] is falsy you will not return True or False, but None which is also falsy.

File: pysma/__init__.py (hunk 2 above)
Comment (kellerza) [reply pysma/__init__.py:335]: Empty key is probably ok if you just need a dashboard view

**Comment by kellerza** [comment]
Thanks @trunone ! Sorry it took ages to merge this

**Comment by trunone** [comment]
No problem, thanks for your work!
